include/envoy/common/scope_tracker.h

```diff
@@ -7,16 +7,19 @@
 namespace Envoy {
 
 /*
- * A class for tracking the scope of work.
- * Currently this is only used for best-effort tracking the L7 stream doing
- * work if a fatal error occurs.
+ * An interface for tracking the scope of work. Implementors of this interface
+ * can be registered to the dispatcher when they're active on the stack. If a
+ * fatal error occurs while they were active, the dumpState method will be
+ * called.
+ *
+ * Currently this is only used for the L4 network connection and L7 stream.
```

Comment:
class ScopeTrackedObject : public Dumpable {}

But are the two interfaces really needed?  Why is it important to distinguish between objects that can be registered and ones that simply have a dump method?

Reply:
Is not being able to register the object useful?  Having 2 interfaces that are so similar adds to cognitive load.

Reply:
Yeah I would tend to agree that the two interfaces do not feel that useful to me, but I don't feel strongly about it. At minimum I would derive from the other class as @antoniovicente suggests.

Reply:
Fair point, will just go with `ScopeTrackedObject` to keep it simple